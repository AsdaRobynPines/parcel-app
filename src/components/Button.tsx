import React from "react";

interface Props {
	className: string;
	children?: React.ReactNode;
	onClick: () => void;
}

const Button: React.FC<Props> = ({ className, children, onClick }) => {
	return (
		<button onClick={onClick} className={className}>
			{children}
		</button>
	);
};

export default Button;
