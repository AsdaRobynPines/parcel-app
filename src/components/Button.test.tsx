import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";

const onClick = jest.fn();

it("renders the button component successfully", () => {
	const component = renderer.create(
		<Button className="button" children="button" onClick={onClick} />
	);
	let testJson = component.toJSON();
	renderer.act(() => testJson.props.onClick());
	expect(testJson).toMatchSnapshot();
	expect(onClick).toHaveBeenCalled();
});
