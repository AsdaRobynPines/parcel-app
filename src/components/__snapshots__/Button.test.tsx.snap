// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button component successfully 1`] = `
<button
  className="button"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  button
</button>
`;
